pkg/systemd/services/service-details.jsx

```diff
@@ -208,11 +208,40 @@ export class ServiceDetails extends React.Component {
             waitsFileAction: false,
             note: "",
             error: "",
+            delay: 30000, // 30s
+            unit_properties: {},
         };
 
         this.onOnOffSwitch = this.onOnOffSwitch.bind(this);
         this.unitAction = this.unitAction.bind(this);
         this.unitFileAction = this.unitFileAction.bind(this);
+        this.addUnitProperties = this.addUnitProperties.bind(this);
+        this.doPropertyPolling = this.doPropertyPolling.bind(this);
+
+        this.unitType = props.unit.Id.split('.').slice(-1)[0];
+        this.unitTypeCapitalized = this.unitType.charAt(0).toUpperCase() + this.unitType.slice(1);
+
+        if (props.unit.ActiveState == "active") {
+            this.doPropertyPolling();
+            this.interval = setInterval(this.doPropertyPolling, this.state.delay);
+        }
```

Comment:
Does this really have to repeat the code in componentDidUpdate()? I thought the latter would be called on initial mounting as well?

Reply:
componentDidUpdate method is not called for the initial render. So when we show the page first time we need to start the polling. Then everytime the page updates, this might mean that the unit changed state, so we might need to start or stop the polling.

Reply:
Ack, thanks; I suppose I was thinking of `useEffect()` as the thing that runs both on initial mount and on update. (That may still be the more elegant way, but let's not block on this).